### PR TITLE
fix(node): don't allow generation to touch samples/

### LINKED ISSUE #757: Synthtool is trying to delete package.json
Looking at this PR here:
https://github.com/googleapis/nodejs-automl/pull/450

It looks like autosynth is trying to delete `samples/package.json`, which is very much needed for the package to operate.  Not sure how it's happening 🤷

### PR DESCRIPTION
**Pull request by bcoe**
synthtool uses the files touched during generation to decide which files to track.

linting/`npm i` sometimes update the files in `samples/`, even though they're not currently generated.

This adds a step that checks out any changes that happen in `samples/` during generation.

_Note: to fix libraries currently in a bad state, we need to remove files in `samples/` from synth.metadata, this will protect us in the future I believe._

Fixes #757

### CODE REVIEW

File: synthtool/languages/node.py [hunk 1]
```diff
@@ -177,6 +178,11 @@ def fix(hide_output=False):
     shell.run(["npm", "run", "prelint"], check=False, hide_output=hide_output)
     logger.debug("Running fix...")
     shell.run(["npm", "run", "fix"], hide_output=hide_output)
+    try:
+        shell.run(["git", "checkout", "samples"], hide_output=hide_output)
```
Comment (SurferJeffAtGoogle) [inline synthtool/languages/node.py:182]: Does this effectively undo any changes that the npm command made?

Comment (bcoe) [reply synthtool/languages/node.py:182]: @SurferJeffAtGoogle yes, it should put just that one folder  back to HEAD.

Note, that it is both npm, and linting that can touch that folder.

**Review by SurferJeffAtGoogle — APPROVED** [review]

**Comment by bcoe** [comment]
I should be more specific and try to checkout `**/*.js`, `**/*.json` perhaps?

**Comment by SurferJeffAtGoogle** [comment]
@bcoe   This pull request has been inactive for over 30 days.  Do you still plan to merge it?

**Comment by SurferJeffAtGoogle** [comment]
Closing because this PR has been dormant for 60+ days.  Feel free to re-open if you intend to revise and merge.

**Pull request closed without merging**